src/clients/detector.rs

```diff
@@ -141,3 +176,23 @@ impl From<ContentAnalysisResponse> for crate::models::TokenClassificationResult
         }
     }
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct GenerationDetectionResponse {
+    /// Relevant detection class
+    pub detection: String,
+    /// Detection type or aggregate detection label
+    pub detection_type: String,
+    /// Score of detection
+    pub score: f64,
+}
+
+impl From<GenerationDetectionResponse> for crate::models::DetectionResult {
```

Comment:
I'm not sure if this trait is needed. At first, I only did that because I was copying from the already existing point. I think this trait could be removed and DetectionResult could be used in both places. However, I'm not sure if this would break any rules in the desired architecture of the code.